operator/api/v1alpha1/resource_types.go

```diff
@@ -32,11 +29,22 @@ type ResourceSpec struct {
 
 // ResourceStatus defines the observed state of Resource
 type ResourceStatus struct {
-	TotalRequirements       int                      `json:"totalRequirements"`
-	ImplementedRequirements int                      `json:"implementedRequirements"`
-	Children                map[string]ResourceChild `json:"children"`
-	TotalChildren           int                      `json:"totalChildren"`
-	CompliantChildren       int                      `json:"compliantChildren"`
+	//+optional
```

Comment:
Why explicitly put a comment here, what about `omitempty`? Is there any other requirement that I'm missing?

This is a real pain for my OCD, seeing as the fields aren't aligned :D

Reply:
These are to tell kubebuilder API to treat the fields as optional when updating status manifests. https://book.kubebuilder.io/cronjob-tutorial/api-design.html. If only omitempty is used, kubernetes validation will refuse the manifest that does not contain the value.

Reply:
I did not understand the misalignment bit :P happy to change that though